src/main/c/com_micron_hse_Kvs.c

```diff
@@ -229,6 +259,76 @@ Java_com_micron_hse_Kvs_get__J_3BIIJ(
     return value;
 }
 
+jbyteArray
+Java_com_micron_hse_Kvs_get__JLjava_lang_String_2IJ(
+    JNIEnv *env,
+    jobject kvs_obj,
+    jlong   kvs_handle,
+    jstring key,
+    jint    flags,
+    jlong   txn_handle)
+{
+    (void)kvs_obj;
+
+    struct hse_kvs      *kvs = (struct hse_kvs *)kvs_handle;
+    struct hse_kvdb_txn *txn = (struct hse_kvdb_txn *)txn_handle;
+
+    jbyteArray  value = NULL;
+    const char *key_data = NULL;
+    jsize       key_len = 0;
+    if (key) {
+        key_data = (*env)->GetStringUTFChars(env, key, NULL);
+        key_len = (*env)->GetStringUTFLength(env, key);
+    }
+
+    jbyte *value_data = malloc(HSE_KVS_VALUE_LEN_MAX);
+    if (!value_data) {
+        (*env)->ThrowNew(
+            env,
+            globals.java.lang.OutOfMemoryError.class,
+            "Failed to allocate memory for value buffer");
+        return NULL;
+    }
```

Comment:
Large mallocs typically end up in a call to mmap which can be very expensive, depends somewhat which allocator you're using...

Reply:
Ideally, this API path is never used, and the user allocates their own buffers. This is purely a convenience function.

Reply:
Gave it some thought. Should I just allocate on the stack? Can't really remember why I chose to allocate on the heap other than maybe avoid a stack overflow.

Reply:
Nevermind this is a huge stack overflow from my testing lol